velox/expression/tests/ConjunctRewriteTest.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include "velox/expression/ExprRewriteRegistry.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+namespace facebook::velox::expression {
+namespace {
+
+class ConjunctRewriteTest : public functions::test::FunctionBaseTest {
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance(memory::MemoryManager::Options{});
+  }
+
+  void SetUp() override {
+    functions::prestosql::registerAllScalarFunctions("");
+    parse::registerTypeResolver();
+  }
+
+  void TearDown() override {
+    expression::ExprRewriteRegistry::instance().clear();
+  }
+
+  /// Validates AND, OR special form expression rewrites that can short circuit
+  /// expression evaluation.
+  /// @param input Input SQL expression to be rewritten.
+  /// @param expected Expected SQL expression after rewriting `input`.
+  /// @param type Type of `input` and `expected`.
+  void testRewrite(
+      const std::string& input,
+      const std::string& expected,
+      const RowTypePtr& type = ROW({})) {
+    const auto expr = makeTypedExpr(input, type);
+    const auto rewritten =
+        expression::ExprRewriteRegistry::instance().rewrite(expr);
+    const auto expectedExpr = makeTypedExpr(expected, type);
+    SCOPED_TRACE(fmt::format("Input: {}", expr->toString()));
+    SCOPED_TRACE(fmt::format("Rewritten: {}", rewritten->toString()));
+    SCOPED_TRACE(fmt::format("Expected: {}", expectedExpr->toString()));
+
+    ASSERT_TRUE(*rewritten == *expectedExpr);
+  }
+};
+
+TEST_F(ConjunctRewriteTest, conjunct) {
```

Comment:
perhaps, add a test to demonstrate that this rewrite enables constant folding that wasn't possible before

something like: if(a > 10 and false, 0, 1) --> 1